### ISSUE TITLE
Exporting return empty images and annotations

### ISSUE THREAD

**Issue by Aaronzinhoo** [jsbroks/coco-annotator#340]
Hi, 

I just imported a coco dataset and tried to export the results after making some adjustments and now the json file is empt. Is there anyway to solve this?

**Comment by Aaronzinhoo** [comment]
I was thinking importing a dataset would at least return the same imported dataset if I do not change anything but this is not the case, it still returns an empty json

**Comment by Aaronzinhoo** [comment]
I was thinking importing a dataset would at least return the same imported dataset if I do not change anything but this is not the case, it still returns an empty json

**Comment by jsbroks** [comment]
If you open an image are the annotations loaded fine?

**Comment by Aaronzinhoo** [comment]
Yes, the annoatations load up on images just fine. The annotations actually are not empty but only contain a couple images instead of 2k. 
The problem occured when I switched from working on the dataset in a production build to a development build. It was a dumb mistake I made since I thought the category filtering was only on the development build. I have tried exporting from both the development build and the production build, as well as deleting the dataset and importing the coco annotations from previous full export I made and still nothing changes. It just exports a few images each time.

**Comment by Aaronzinhoo** [comment]
The categories and statistics also do not match each other. Maybe i should make a new account since my profile seems to be corrupted sadly. Sorry to bother with this and thanks for any help you can provide.

**Comment by Aaronzinhoo** [comment]
Tried to use 
```
docker volume rm --force cocoannotator_mongodb_data
```
But all it did stop the export from having any annotations or images. Even though I cleared the database the categories are still all full. Should I just create a new account?

**Comment by dreaquil** [comment]
I think this is similar to #335. If you find a solution please let us know.

**Comment by Aaronzinhoo** [comment]
Yes, I will. Unfortunately I do not think I have time to find a solution and have tried many different things. Ill look at #335 and see if I can go off of what they discussed

**Comment by Aaronzinhoo** [comment]
@dreaquil after trying #335 suggestion of taking num_annotation out of loop, nothing changed. The images and the annotations from the export are still blank unfortunately. Since my dataset was also deleted I may have to try another sadly. I really do love what yall did overall with this program!

**Comment by Aaronzinhoo** [comment]
@jsbroks @dreaquil I was able to get the export working again by reverting the repo to the provious version. We may be able to localize where the break occurs by trying to export starting from this point. 
```
git checkout  aa78c426fe98ad948aa85048b76508d4f34d10c5 . 
```

**Comment by dreaquil** [comment]
@Aaronzinhoo Thanks for getting back with the update. I just tried the branch but I still get the same issue for some reason. It only exports the images I've cycle through the annotation tool view. Was there anything else you tried?

**Comment by Aaronzinhoo** [comment]
@dreaquil Hmm, well i also cleared the dataset volumes from mongodb. These bugs may be unrelated actually. For my issue, the export would be empty or at least a consistent (low) amount of images but annotations.

**Comment by dreaquil** [comment]
@Aaronzinhoo I went for a fresh clone of the git repo and imported the dataset again on the branch but still got the same. So you're probably right that the bugs may be unrelated. Can I ask what size of dataset you're using?

**Comment by jsbroks** [comment]
In #338 I explain how to revert using docker. I dont gave time atm to fix this, hopefully next week

**Comment by Aaronzinhoo** [comment]
@jsbroks @dreaquil sounds good! Hopefully that works for @dreaquil's bug. Also I was working with 5000 images @dreaquil but only 2000 are annotated.

**Comment by dreaquil** [comment]
@Aaronzinhoo @jsbroks I think I found a fix for my issue and perhaps yours too. 

In `backend/worker/tasks/data.py`, for some reason the image_model update on line 297 isn't working properly and the annotated flag isn't being set. I added `image_model.update(set__annotated=True)` to make sure it is applied and the import and exporter are both working fine now.

**Comment by Aaronzinhoo** [comment]
@dreaquil Ill try it out tonight!  Should definitely be a PR if that fixes it. @jsbroks

**Comment by dreaquil** [comment]
@Aaronzinhoo any update?

**Comment by timofurrer** [comment]
I have the exact same issue. Downloading from the `Datasets` overview page via the three dot menu and `Download COCO` works though as a workaround!

**Comment by Aaronzinhoo** [comment]
So @dreaquil @jsbroks  I have been working on other project since I was able to that portion to work and train my dataset. The startup I work in moves pretty fast. I have not gotten to check it yet, but I will this weekend since I cannot leave my house anyways

**Comment by Aaronzinhoo** [comment]
@timofurrer it works, but the issue is if you had some classes that you don't want to use in your dataset, you will have to extract them manually.

**Comment by timofurrer** [comment]
@Aaronzinhoo Yes, that's correct - but that isn't too much of a problem .. 
The [COCO data format](http://cocodataset.org/#format-data) documentation gives you pretty good start for that ;)
